weblate/api/serializers.py

```diff
@@ -113,6 +114,89 @@ def validate_code(self, value):
         return value
 
 
+class UserSerializer(serializers.ModelSerializer):
+    groups = serializers.HyperlinkedIdentityField(
+        view_name="api:group-detail", lookup_field="name", many=True, read_only=True,
+    )
+
+    class Meta:
+        model = User
+        fields = (
+            "email",
+            "full_name",
+            "username",
+            "groups",
+            "is_superuser",
+            "is_active",
+            "date_joined",
+            "url",
+        )
+        extra_kwargs = {
+            "url": {"view_name": "api:user-detail", "lookup_field": "username"}
+        }
+
+
+class PermissionSerializer(serializers.RelatedField):
+    class Meta:
+        model = Permission
+
+    def to_representation(self, value):
+        return value.name
+
+
+class RoleSerializer(serializers.ModelSerializer):
+    permissions = PermissionSerializer(read_only=True, many=True)
+
+    class Meta:
+        model = Role
+        fields = (
+            "name",
+            "permissions",
+        )
+
+    def validate_name(self, value):
+        check_query = Role.objects.filter(name=value)
+        if not check_query.exists():
+            raise serializers.ValidationError("Role with this role name was not found.")
+        return value
+
+
+class GroupSerializer(serializers.ModelSerializer):
+    roles = RoleSerializer(many=True)
+
+    class Meta:
+        model = Group
+        fields = (
+            "name",
+            "roles",
+            "project_selection",
+            "language_selection",
+            "url",
+        )
+        extra_kwargs = {
+            "url": {"view_name": "api:group-detail", "lookup_field": "name"}
```

Comment:
I'm not sure about this. The group name can contain any chars and using that in URL might be problematic...

Reply:
That's true. I will update this to use ID instead then. And also for the association with users.

Reply:
For users the username is already used in URLs, so that should be safer in this regard.